### PR TITLE
Add structure and ui tipps&tricks

### PR DESCRIPTION
**Pull request by LasseDenecken**
👋 @TwitchBronBron I'm not 100% sure where you're going with this guide - but i already love it!
I just added some quick thinks, that came to my mind and would have been helpful to know once I started developing roku channels. Let me know if you need me to go into more detail / add code examples or missed the gist of the roku-guide completely.

### CODE REVIEW

File: docs/Structure/singleton-and-globals.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Singleton
+For some use-cases you want to always have the same instance of a component. F.e. when using app wide repositories oder managers.
```
Comment (TwitchBronBron) [inline docs/Structure/singleton-and-globals.md:2]: spelling mistake? `oder`

File: docs/UI/round-corners.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Round corners
+There is no native way to set a corner-radius for components / rectangle / anything at all.
+The best results can be achieved by using a MaskGroup. Keep in mind that MaskGroups can't use 9-patch-images.
```
Comment (TwitchBronBron) [inline docs/UI/round-corners.md:3]: Could you add a code example of how to implement rounded corners using a `MaskGroup`?

File: docs/UI/shadows.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Shadows
+There is no native way to add shadows to a component / rectangle / anything at all.
+You have to fallback on adding shadows as images. If you are using similar shadows in your UI you can make use of 9-patch-images.
```
Comment (TwitchBronBron) [inline docs/UI/shadows.md:3]: How would you use a 9-patch to add shadows?

**Comment by TwitchBronBron** [comment]
We're not totally sure where this is going either. 😄 For now, we're just adding all the stuff we can think of, and I figure we can format and tweak it as we go. I've been imagining a full step-by-step tutorial for many different steps, but that'll take some time, so I think for now this is mostly just a suite of "best practices" and gotchas.

**Review by TwitchBronBron — CHANGES_REQUESTED** [review]

**Review by LasseDenecken — COMMENTED** [review]
Fixed the spelling mistake (_a German word slipped in there_) and added a code example.

**Review by TwitchBronBron — APPROVED** [review]

**Pull request merged**